### ISSUE TITLE
How do I modify the code to run other datasets?

### ISSUE THREAD

**Issue by zhangyuanyang21** [DanielTrosten/mvc#4]
Hi Daniel J. Trosten, your SiMVC and CoMVC is very good, I am very interested in SiMVC and CoMVC, how can SiMVC and CoMVC network run other datasets (e.g. datasets in .mat form, such as 100leaves.mat,Caltech-5V.mat)? ), is it possible to run large scale datasets? (e.g. cifar10.mat, cifar100.mat, YoutubeFace_sel_fea.mat), how do I need to modify the code? I look forward to hearing from you! Good luck with your work and happy life!

**Comment by DanielTrosten** [comment]
Hello, and thank you for taking interest in our work!
 
It should be fine to run on any dataset (though large-scale might take some time), as long as you format it as specified in the README. I.e. an npz file with the appropriate elements:

------
### Preparing a custom dataset for training
Create `<custom_dataset_name>.npz` in `data/processed/` with the following keys:
```
n_views: The number of views, V
labels: One-dimensional array of labels. Shape (n,)
view_0: Data for first view. Shape (n, ...)
  .
  .
  .
view_V: Data for view V. Shape (n, ...)
```
Alternatively, call
```Python
data.make_dataset.export_dataset(
    "<custom_dataset_name>",    # Name of the dataset
    views,                      # List of view-arrays
    labels                      # Label array
)
```
This will automatically export the dataset to an `.npz` file at the correct location.

Then, in the Experiment-config, set
```Python
dataset_config=Dataset("<custom_dataset_name>")
```

## Experiment configuration
Experiment configs are nested configuration objects, where the top-level config is an instance of 
`config.defaults.Experiment`. 

The configuration object for the contrastive model on E-MNIST, for instance, looks like this:
```Python
from config.defaults import Experiment, CNN, DDC, Fusion, Loss, Dataset, CoMVC, Optimizer


mnist_contrast = Experiment(
    dataset_config=Dataset(name="mnist_mv"),
    model_config=CoMVC(
        backbone_configs=(
            CNN(input_size=(1, 28, 28)),
            CNN(input_size=(1, 28, 28)),
        ),
        fusion_config=Fusion(method="weighted_mean", n_views=2),
        projector_config=None,
        cm_config=DDC(n_clusters=10),
        loss_config=Loss(
            funcs="ddc_1|ddc_2|ddc_3|contrast",
            # Additional loss parameters go here
        ),
        optimizer_config=Optimizer(
            learning_rate=1e-3,
            # Additional optimizer parameters go here
        ) 
    ),
    n_epochs=100,
    n_runs=20,
)
```

**Comment by zhangyuanyang21** [comment]
Thank you very much for your help!

**Comment by zhangyuanyang21** [comment]
Then, in the Experiment-config, set
dataset_config=Dataset("<custom_dataset_name&gt;")



Where is “dataset_config=Dataset("<custom_dataset_name&gt;")”, please? I can't find the corresponding location in the code.

How should I do this part?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "DanielTrosten/mvc"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年9月13日(星期二) 下午3:29
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [DanielTrosten/mvc] How do I modify the code to run other datasets? (Issue #4)





 
Hello, and thank you for taking interest in our work!
 
It should be fine to run on any dataset (though large-scale might take some time), as long as you format it as specified in the README. I.e. an npz file with the appropriate elements:
 
 
Preparing a custom dataset for training
 
Create <custom_dataset_name&gt;.npz in data/processed/ with the following keys:
 n_views: The number of views, V labels: One-dimensional array of labels. Shape (n,) view_0: Data for first view. Shape (n, ...)   .   .   . view_V: Data for view V. Shape (n, ...)  
Alternatively, call
 data.make_dataset.export_dataset(     "<custom_dataset_name&gt;",    # Name of the dataset     views,                      # List of view-arrays     labels                      # Label array )
 
This will automatically export the dataset to an .npz file at the correct location.
 
Then, in the Experiment-config, set
 dataset_config=Dataset("<custom_dataset_name&gt;")
 
Experiment configuration
 
Experiment configs are nested configuration objects, where the top-level config is an instance of
 config.defaults.Experiment.
 
The configuration object for the contrastive model on E-MNIST, for instance, looks like this:
 from config.defaults import Experiment, CNN, DDC, Fusion, Loss, Dataset, CoMVC, Optimizer mnist_contrast = Experiment(     dataset_config=Dataset(name="mnist_mv"),     model_config=CoMVC(         backbone_configs=(             CNN(input_size=(1, 28, 28)),             CNN(input_size=(1, 28, 28)),         ),         fusion_config=Fusion(method="weighted_mean", n_views=2),         projector_config=None,         cm_config=DDC(n_clusters=10),         loss_config=Loss(             funcs="ddc_1|ddc_2|ddc_3|contrast",             # Additional loss parameters go here         ),         optimizer_config=Optimizer(             learning_rate=1e-3,             # Additional optimizer parameters go here         )      ),     n_epochs=100,     n_runs=20, )
